2-Authorization-I/1-call-graph/SPA/src/hooks/useTokenAcquisition.js

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from 'react';
+import { InteractionRequiredAuthError, InteractionStatus, InteractionType } from '@azure/msal-browser';
+import { useMsal } from '@azure/msal-react';
+
+/**
+ * A custom hook for acquiring tokens using MSAL
+ * @param {Array} scopes
+ * @param {String} interactionType
+ * @returns response object which contains an access token
+ */
+const useTokenAcquisition = (scopes, interactionType) => {
+    /**
+     * useMsal is a hook that returns the PublicClientApplication instance,
+     * an array of all accounts currently signed in and an inProgress value
+     * that tells you what msal is currently doing. For more, visit:
+     * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md
+     */
+
+    const { instance, inProgress } = useMsal();
+    const account = instance.getActiveAccount();
+    const [response, setResponse] = useState(null);
+
+    useEffect(() => {
+        const getToken = async () => {
+            let token;
+            if (account && inProgress === InteractionStatus.None && !response) {
+                try {
+                    token = await instance.acquireTokenSilent({
```

Comment:
```suggestion
                    token = await instance.acquireTokenSilent({
```

`response` or `tokenResponse` would be more accurate